core/request.go

```diff
@@ -104,14 +104,23 @@ func (r *RequestAccessor) StripBasePath(basePath string) string {
 	return newBasePath
 }
 
-// ProxyEventToHTTPRequest converts an API Gateway proxy event into an
-// http.Request object.
+// ProxyEventToHTTPRequest converts an API Gateway proxy event into a http.Request object.
 // Returns the populated request with:
-// 		* an additional two custom headers for the stage variables and API Gateway context.
-//      To access these properties use the GetAPIGatewayStageVars and GetAPIGatewayContext method of the RequestAccessor object.
-//    * lambda runtime context passed as well as APIGatewayProxyRequestContext as part of it's context under different keys.
-//      Access those using GetAPIGatewayContextFromContext and GetRuntimeContextFromContext methods in this package.
-func (r *RequestAccessor) ProxyEventToHTTPRequest(ctx context.Context, req events.APIGatewayProxyRequest) (*http.Request, error) {
+// * an additional two custom headers for the stage variables and API Gateway context.
+//   To access these properties use the GetAPIGatewayStageVars and GetAPIGatewayContext method of the RequestAccessor object.
+// * lambda context passed as well as APIGatewayProxyRequestContext as part of its context under different keys.
+//   Access those using GetAPIGatewayContextFromContext and GetRuntimeContextFromContext functions in this package.
+func (r *RequestAccessor) ProxyEventToHTTPRequest(req events.APIGatewayProxyRequest) (*http.Request, error) {
+	return r.RequestFromEvent(context.Background(), req)
+}
+
+// RequestFromEvent converts an API Gateway proxy event and context into an http.Request object.
+// Returns the populated request with:
+// * an additional two custom headers for the stage variables and API Gateway context.
```

Comment:
You could use this opportunity to make the _new_ method not modify http headers: it would just include the request context. The old method could then just call RequestFromEvent, and as a post-processing step, add those two headers. This makes the new method have clean behavior from the start, with the old method adding the headers solely for backwards compatibility.